### PR TITLE
Merge nodejs/master

### PR DESCRIPTION
**Pull request by kunalspathak**
Merge 3e178848a5 as of 2017-06-22.

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

core, n-api, test

### CODE REVIEW

File: src/node_api_jsrt.cc [hunk 1]
```diff
@@ -348,16 +348,17 @@ const char* error_messages[] = {
...
+  "napi_escape_handle already called on scope"
 };
 
 napi_status napi_get_last_error_info(napi_env env,
                                      const napi_extended_error_info** result) {
   CHECK_ARG(result);
 
   static_assert(
-    sizeof(error_messages) / sizeof(*error_messages) == napi_status_last,
+    node::arraysize(error_messages) == napi_escape_called_twice + 1,
```
Comment (kfarnung) [inline src/node_api_jsrt.cc:359]: There's an upstream PR to eliminate `node_internal.h` from `node_api.h`, we should probably avoid taking any further dependency on it.

https://github.com/nodejs/node/pull/13892

**Review by kfarnung — APPROVED** [review]
LGTM with minor changes.

File: src/node_api_jsrt.cc (hunk 1 above)
Comment (kfarnung) [reply src/node_api_jsrt.cc:359]: Although taking another look it seems like there's some discussion ongoing, so we can leave this for now.

Comment (kunalspathak) [reply src/node_api_jsrt.cc:359]: Yeah, i just adopted this changes from `node_api.cc`.

Comment (kfarnung) [reply src/node_api_jsrt.cc:359]: Yeah, that PR was closed now and we're going a different direction.  LGTM as is.

**Review by kfarnung — APPROVED** [review]

**Pull request merged**